### PR TITLE
Implement locked keybindings

### LINKED ISSUE #2034: [Feature] Allow certain keybinds to be used during a locked state
As the title suggests, it would be useful to allow certain keybinds (volume and media keys to name a few) to be used while the session is locked.

### PR DESCRIPTION
**Pull request by xi**
fixes #2034

There is a [similar feature in sway](https://github.com/swaywm/sway/blob/bc258a3be2f946c1c93bcbe40735b2db068e0ea8/sway/sway.5.scd?plain=1#L409).

### CODE REVIEW

File: src/config/keybind.c [hunk 1]
```diff
@@ -120,6 +120,7 @@ keybind_create(const char *keybind)
 {
 	xkb_keysym_t sym;
 	struct keybind *k = znew(*k);
+	k->locked = false;
```
Comment (Consolatis) [inline src/config/keybind.c:123]: This isn't needed, `znew()` will zero out all memory which means it default to `false` anyway.

**Comment by Consolatis** [comment]
Code looks good to me apart from the comment above (the CI static sanitizer is a bit stupid, `k` may be set to `NULL`).
Didn't test.

I don't really have an opinion if we want the feature itself or not although I think it makes sense to e.g. control volume buttons and similar while locked.

CC @ahesford @jlindgren90 @johanmalm @tokyo4j

**Comment by jlindgren90** [comment]
+1 to the idea, this would be useful to me for volume and media keys.

**Comment by ahesford** [comment]
The idea isn't bad, but I don't like the keyword "locked" because it has a bunch of competing connotations that make the intent non-obvious.

Something like "allowWhenLocked" would be clearer.

**Comment by tokyo4j** [comment]
I like this feature, but you should also modify the logic for key release:
```diff
diff --git a/src/input/keyboard.c b/src/input/keyboard.c
index 0caaf406..3529f410 100644
--- a/src/input/keyboard.c
+++ b/src/input/keyboard.c
@@ -458,7 +458,7 @@ handle_compositor_keybindings(struct keyboard *keyboard,
 	if (event->state == WL_KEYBOARD_KEY_STATE_RELEASED) {
 		if (cur_keybind && cur_keybind->on_release) {
 			key_state_bound_key_remove(event->keycode);
-			if (seat->server->session_lock_manager->locked) {
+			if (locked && !cur_keybind->allowWhenLocked) {
 				cur_keybind = NULL;
 				return true;
 			}

```

File: include/config/keybind.h [hunk 2]
```diff
@@ -18,6 +18,7 @@ struct keybind {
 	xkb_keycode_t keycodes[MAX_KEYCODES];
 	size_t keycodes_len;
 	int keycodes_layout;
+	bool allowWhenLocked;
```
Comment (tokyo4j) [inline include/config/keybind.h:21]: nit: we use pascal case in `rc.xml`, but use snake case for variable names.
```suggestion
	bool allow_when_locked;
```

**Comment by Consolatis** [comment]
Tested and works as expected. Thanks!

**Pull request merged**